### ISSUE TITLE
ChromeOS: git repo loses many files

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#950]
Found during https://github.com/kinu-garage/hut_10sqft/issues/932

# issue
Needs more issue isolation though, what I'm seeing is:
- A local git repo lost many files but not all.
   - I noticed this after OS reboot.
   - On the synched folder on G-Drive, I do see the same files/folderes are missing. Wtf.
- The git repo sits on a path that synchs with Google Drive (by ChromeOS' feature).
- At a glance the files lost and found:
   - Found: Files, a hidden directory.
   - Lost: Directories.
   ```
   $ ls -alth
   total 29K
   drwxr-x--- 3 noodle-crafter chronos-access 4.0K Dec  6 07:31 .git
   drwxr-x--- 8 noodle-crafter chronos-access 4.0K Nov 29 06:10 ..
   drwxr-x--- 3 noodle-crafter chronos-access 4.0K Jun 27 06:27 .
   -rw-r----- 1 noodle-crafter chronos-access 1.6K Mar 30  2023 package.xml
   -rw-r----- 1 noodle-crafter chronos-access  788 Mar 28  2019 .travis.yml
   -rw-r----- 1 noodle-crafter chronos-access  544 Aug 14  2017 .gitignore
   -rw-r----- 1 noodle-crafter chronos-access  12K Aug 14  2017 LICENSE
   -rw-r----- 1 noodle-crafter chronos-access  666 Aug 14  2017 README.rst
   ```

**Comment by 130s** [comment]
The files found in G-Drive's Trash shows the timestamp of 12/6.
5 days ago!? I don't remember what I did back that long time ago, or maybe that was where I was trying to resolve https://github.com/kinu-garage/hut_10sqft/issues/932 by removing the local repo and re-copy-pasting the repo.

Anyways, doing the same again --
1. [x] removing and re-pasting the local repo,
1. [x] reboot the OS (b/c OS update is available).
1. [ ] If the same issue happens, maybe the issue is particular to OS reboot? Re-paste again.
1. [ ] keep an eye on whether the local repo loses some folders over the time.
1. [ ] reboot the OS.

**Comment by 130s** [comment]
!!! After re-copy-pasting the local repo,
1. I waited for the local repo to finish synching. More accurately, I waited for the `Files` app's pop-up notification that says synching on G-Drive (I verified this is not the one from Dropbox).
1. From `bash`, I do not see any file including the repo directory itself at the path.
1. On a new `bash` terminal, I do see all the files as expected!
1. `git {log, status}` shows as expected!

Now rebooting the OS (installing OS updates) and see.

**Comment by 130s** [comment]
Rebooted and I still see reasonably many files and `git` still works!
I can't think of a way to confirm what happened. I conclude that after manual deletion of the local repo, somehow the newly pasted files didn't retain in the path.

Moved to self-review.

**Comment by 130s** [comment]
Closing for now.

The issue claimed in the title doesn't seem to be happening. So for now I can only conclude that there was my human-mistake, could be either:
- I removed some files.
- After initiated copying a folder on to the dest. path, I didn't wait long enough, based on my observation that ChromeOS seems to take longer time to complete copy operation between certain paths. Pair of paths where I've observed it so far (This, combined with Dropbox path not being able to mount on to `/mnt` in Linux mode https://github.com/kinu-garage/hut_10sqft/issues/891#issuecomment-1722391388, gives me an impression that the file systems / logical disks are separate per each location in the pair below):
   - somewhere in "`My files`" -> somewhere in "`Google Drive`"
   - somewhere in "`My files`" -> somewhere in "`Dropbox`"
   - somewhere in "`Google Drive`" -> somewhere in "`Dropbox`"
